doc/toolchains/llvm.chapter.md

```diff
@@ -0,0 +1,31 @@
+# The LLVM Toolchain {#chap-toolchains}
+
+LLVM is a target-independent optimizer and code generator and serves as the basis for many compilers such as Haskell's GHC, Rust, Zig, and many others. It forms the base tools for Apple's Darwin platform.
+
+## Using LLVM {#sec-using-llvm}
+
+LLVM has two ways of being used. One is by using it across all of nixpkgs and the other is to compile and build other packages.
+
+### Building packages with LLVM {#sec-building-packages-with-llvm}
+
+Nixpkgs supports two methods of compiling the world with LLVM, one is via setting `useLLVM` in `crossSystem` while importing. This is the recommended way when doing cross compiling as it is more expressive. An example of doing `aarch64-linux` cross from `x86_64-linux` with LLVM on the target is by doing the following:
+
+```nix
+import <nixpkgs> {
+  localSystem = {
+    system = "x86_64-linux";
+  };
+  crossSystem = {
+    useLLVM = true;
+    linker = "lld";
+  };
+}
+```
+
+Note that we set `linker` to `lld`, this is because LLVM has its own linker called "lld". By setting it, we utilize Clang and lld within this new instance of nixpkgs. The other method which nixpkgs supports for building everything with LLVM is via `pkgsLLVM`. This is recommended to be used with `nix-build` (or `nix build`). An example of this is as simple as the following:
```

Comment:
I'm not sure I would describe pkgsLLVM as another method, since it's the same thing. Calling it a shorthand would be better.  I think "easier to use" would be better than "recommended to be used also".

Reply:
Ok, I've changed it.